Add Failover service (#8825)
traefik/traefik: docs/content/routing/services/index.md

        url = "http://private-ip-server-2/"
```

#### FailoverService

> Failover service allows to reference a service that will be use when no servers are available on the loadbalancer.

Reviewer: ```suggestion
Failover service allows referencing a service that will be used when no servers are available on the loadbalancer.
```

> !!! info "This is having effects only when `healthCheck` is set on children"

Reviewer: ```suggestion
!!! info "A failover service can only take over when all the children of the initial service are detected as unavailable. This can only happen if `healthCheck` is enabled for all the children."
```

        url = "http://private-ip-server-2/"
```

### Failover (service)

The failover is able to forward requests sent to a main service to a fallback service when the main service becomes unreachable. 

> !!! info "Supported Providers"

Reviewer: ```suggestion
!!! info "Relation to HealthCheck"
    The failover service relies on the HealthCheck system to get notified when its main service becomes unreachable, which means HealthCheck needs to be enabled and functional on the main service. However, HealthCheck does not need to be enabled on the failover service itself for it to be functional. It is only required in order to propagate upwards the information when the failover itself becomes down (i.e. both its main and its fallback are down too).

!!! info "Supported Providers"
```

```yaml tab="YAML"
## Dynamic configuration
http:
  services:
    app:
      failover:
        service: main
        fallback: backup
        
    main:
      loadBalancer:
        healthCheck:
          path: /status
          interval: 10s
          timeout: 3s
        servers:
        - url: "http://private-ip-server-1/"

    backup:
      loadBalancer:
        servers:
        - url: "http://private-ip-server-2/"
```

```toml tab="TOML"
## Dynamic configuration
[http.services]
  [http.services.app]
    [http.services.app.failover]
      service = "main"
      fallback = "backup"

  [http.services.main]
    [http.services.main.loadBalancer]
      [http.services.main.loadBalancer.healthCheck]
        path = "/health"
        interval = "10s"
        timeout = "3s"
      [[http.services.main.loadBalancer.servers]]
        url = "http://private-ip-server-1/"

  [http.services.backup]
    [http.services.backup.loadBalancer]
      [[http.services.backup.loadBalancer.servers]]
        url = "http://private-ip-server-2/"
```

#### Health Check

HealthCheck enables automatic self-healthcheck for this service, 

> i.e. if the main and the fallback handlers of the service becomes unreachable, 

Reviewer: ```suggestion
i.e. if the main and the fallback handlers of the service become unreachable, 
```

        url = "http://private-ip-server-2/"
```

### Failover (service)

The failover is able to forward requests sent to a main service to a fallback service when the main service becomes unreachable. 

!!! info "Relation to HealthCheck"

>     The failover service relies on the HealthCheck system to get notified when its main service becomes unreachable, which means HealthCheck needs to be enabled and functional on the main service. However, HealthCheck does not need to be enabled on the failover service itself for it to be functional. It is only required in order to propagate upwards the information when the failover itself becomes down (i.e. both its main and its fallback are down too).

Reviewer: ```suggestion

    The failover service relies on the HealthCheck system to get notified when its main service becomes unreachable, which means HealthCheck needs to be enabled and functional on the main service. However, HealthCheck does not need to be enabled on the failover service itself for it to be functional. It is only required in order to propagate upwards the information when the failover itself becomes down (i.e. both its main and its fallback are down too).
```
